src/utils/issue.ts

```diff
@@ -0,0 +1,200 @@
+  let issueComments: IssueComments | undefined = comments;
+  const linkedIssues: {
+    issueNumber: number;
+    repo: string;
+  }[] = [];
+
+  if (!issueComments && !params) {
+    throw new Error("Either issueComments or params must be provided");
+  }
+
+  if (!issueComments) {
+    issueComments = await fetchIssueComments(params);
+  }
+
+  const {
+    context: {
+      logger,
+      payload: {
+        repository: {
+          owner: { login },
+        },
+      },
+    },
+  } = params;
+
+  if (!issueComments) {
+    logger.info("No comments found on issue");
+    return linkedIssues;
+  }
+
+  for (const comment of issueComments) {
+    const linkedIssue = idIssueFromComment(login, comment.body);
+    if (linkedIssue) {
+      linkedIssues.push(linkedIssue);
+    }
+  }
+
+  return await filterLinkedIssues(params, linkedIssues);
+}
+
+async function filterLinkedIssues(params: FetchParams, linkedIssues: { issueNumber: number; repo: string }[]) {
+  const {
+    context: {
+      logger,
+      payload: {
+        repository: {
+          owner: { login },
+        },
+      },
+    },
+  } = params;
+
+  const contextIssues: {
+    issueNumber: number;
+    repo: string;
+  }[] = [];
+
+  for (const issue of linkedIssues) {
+    if (issue && issue.issueNumber && issue.repo) {
+      if (await isRepoFromSameOrg(params.context, issue.repo, login)) {
+        contextIssues.push({
+          issueNumber: issue.issueNumber,
+          repo: issue.repo,
+        });
+      } else {
+        logger.info(`Ignoring linked issue ${issue.issueNumber} from ${issue.repo} as it is not from the same org`);
+      }
+    }
+  }
+
+  return contextIssues;
+}
+
+export async function getLinkedIssueContextFromComments(context: Context, issueComments: IssueComments) {
+  // find any linked issues in comments by parsing the comments and enforcing that the
+  // linked issue is from the same org that the current issue is from
+  const linkedIssues = await fetchLinkedIssues({ context }, issueComments);
+
+  // the conversational history of the linked issues
+  const linkedIssueComments: IssueComments = [];
+
+  // we are only going one level deep with the linked issue context fetching
+  for (const issue of linkedIssues) {
+    const fetched = await fetchIssueComments({ context, issueNum: issue.issueNumber, repo: issue.repo });
+    linkedIssueComments.push(...fetched);
+  }
+
+  return { linkedIssues, linkedIssueComments };
+}
+
+export function idIssueFromComment(owner?: string, comment?: string | null) {
+  if (!comment) {
+    return null;
+  }
+  if (!owner) {
+    throw new Error("Owner must be provided when parsing linked issues");
+  }
+  // the assumption here is that any special GitHub markdown formatting is converted to an anchor tag
+  const urlMatch = comment.match(/https:\/\/github.com\/([^/]+)\/([^/]+)\/(pull|issue|issues)\/(\d+)/);
+
+  /**
+   * I think we should restrict including any linked context which is not of the same org.
```

Comment:
I think that recursion is not a problem due to generous context length for 4o

Reply:
I'm of a different mindset especially when linking to known 3rd party issues for popular packages which can often have hundreds of comments none of which are truly relevant other the what our commenter had to say before linking the issue.

Like you always say signal over noise, I don't feel it's a benefit and will cause more issues than it will solve. But I'll remove this restriction and see how it fairs.

Reply:
I prefer maximizing signal to noise, this is true in the context of human communication. 
 
These LLMs on the other hand work their magic by dealing with a ton of data. I've seen better results when providing more context than less.